Missing `python3-dev` dependency

**Describe the bug**

Failure to install Python dependencies due to missing `python3-dev` package. Package is not mentioned in [Install dependencies](https://docs.zephyrproject.org/latest/getting_started/index.html#install-dependencies).

**To Reproduce**

1. Install Ubuntu 20.04 LTS on a machine.
2. Follow [Getting Started Guide](https://docs.zephyrproject.org/latest/getting_started/index.html#getting-started-guide)
3. Fail on last step of [Get Zephyr and install Python dependencies](https://docs.zephyrproject.org/latest/getting_started/index.html#get-zephyr-and-install-python-dependencies)

**Expected behavior**

No errors after `pip3 install --user -r ~/zephyrproject/zephyr/scripts/requirements.txt`.

**Impact**

Annoyance. Solved via `sudo apt install python3-dev`.

**Screenshots or console output**

```
  ...
  psutil/_psutil_common.c:9:10: fatal error: Python.h: No such file or directory
      9 | #include <Python.h>
        |          ^~~~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for psutil1
  ...
```

**Environment (please complete the following information):**

 - OS: Ubuntu 20.04 LTS
 - Toolchain: N/A
 - Commit SHA: `7778c22b6031ae766df7cd08522eb9675a0aeb69`

--- a/doc/getting_started/index.rst
+++ b/doc/getting_started/index.rst
@@ -63,7 +63,7 @@ Next, you'll install some host dependencies using your package manager.
 
             sudo apt install --no-install-recommends git cmake ninja-build gperf \
               ccache dfu-util device-tree-compiler wget \
-              python3-pip python3-setuptools python3-tk python3-wheel xz-utils file \
+              python3-dev python3-pip python3-setuptools python3-tk python3-wheel xz-utils file \
               make gcc gcc-multilib g++-multilib libsdl2-dev
 
       #. Verify the version of cmake installed on your system using::

--- a/doc/getting_started/installation_linux.rst
+++ b/doc/getting_started/installation_linux.rst
@@ -78,7 +78,7 @@ need one.
 
          sudo apt-get install --no-install-recommends git cmake ninja-build gperf \
            ccache dfu-util device-tree-compiler wget \
-           python3-pip python3-setuptools python3-tk python3-wheel xz-utils file \
+           python3-dev python3-pip python3-setuptools python3-tk python3-wheel xz-utils file \
            make gcc gcc-multilib g++-multilib libsdl2-dev
 
    .. group-tab:: Fedora

